### ISSUE TITLE
General Discussion Thread

### ISSUE THREAD

**Issue by 64bitpandas** [FewdpewGames/unity-game#10]
**For lack of a better way of doing this, comment general stuff in this thread.**
If you have problems or want to show something off, just take a screenshot and paste it here!

**Comment by 64bitpandas** [comment]
https://github.com/BenCuan/unity-game/wiki/Workflow-Management

**Comment by 64bitpandas** [comment]
![image](https://cloud.githubusercontent.com/assets/15005763/12954952/aa687f0e-cfd5-11e5-944c-30625512d44a.png)

The map is coming along nicely.

**Comment by 64bitpandas** [comment]
Check out the website!

http://fewdpew.me

**Comment by DarkFlamex1** [comment]
Nice alpha ;)

**Comment by EnumC** [comment]
Everyone who is editing the code, if you are thinking of building(Compiling) the code, don't forget to select the "Mac Build Support" And "Linux Build Support" during Unity installation. You can't add it afterwards, unless you install the whole thing again. Just a note. :notebook_with_decorative_cover:  I learned it the hard way :cry:

**Comment by 64bitpandas** [comment]
Also check out the "Update Time" wiki page to see the list of everything
you need to do during an update. I'll probably manage most updates, but
just to let you know how it's done :)

**Comment by 64bitpandas** [comment]
We moved this repository to an organization! Please run this command in the repo: 
git remote set-url origin https://github.com/FewdpewGames/unity-game

If you don't have permission to the repo/organization, please reply here and we'll grant access to you.

**Comment by DarkFlamex1** [comment]
Wait did Eric Q tell you about me?

**Comment by BenCuan** [comment]
Kind of/not really.

**Comment by DarkFlamex1** [comment]
Ok... Well Eric Q can't give me a straight answer, but do you guys want to use a nodejs server for the game, so people can run it on there own machines or photon server or PUN which where photon hosts the server, we can also use a java server instead of nodejs....

**Comment by BenCuan** [comment]
I'm not sure, since I don't know much about networking/multiplayer, and
Eric is managing that stuff (I think). I'll try asking him again.

**Comment by DarkFlamex1** [comment]
Forget everything i said :) We can use PUN

**Comment by DarkFlamex1** [comment]
Um I am working on the server connection stuff but before I start I got some questions:
1)When you start up the game and press play game should there be a list of "rooms" to choose from and be able to create a room? Or just use matchmaking...

**Comment by 64bitpandas** [comment]
I think we will just have a few rooms to start with, then switch to
matchmaking later in development.

**Comment by DarkFlamex1** [comment]
ok

**Comment by 64bitpandas** [comment]
Guys,
Remember to update unity to 5.3.4! If some of us are using 5.3.2 and others using 5.3.4 then we will have compatibility issues.

**Comment by Eric1084** [comment]
And if you are helping with development, but aren't in the group chat already, leave a comment here to be added. We will give you some more information about the Unity Editor.

**Comment by 64bitpandas** [comment]
We need a better way to integrate git into this unity project.

My proposal:
Commit ONLY scripts & other individual files. Do NOT edit any scene files
(.unity)!
Then, in the detailed commit comment (example of how to do this: git commit
-m"basic message" -m"This is a really long
message.....askjdfksjdkgshglkds") comment on how to integrate your files
into the scene (example: "drag onto player", "drag the greenTransform into
the script transform variable", etc.)

I will edit the scene and integrate all these commits together every few
days.

**Comment by DarkFlamex1** [comment]
I don't know about that cause I built a seperate scene for the entire networking thing

**Comment by DarkFlamex1** [comment]
Oh wait nvm

**Comment by DarkFlamex1** [comment]
Hmm

**Comment by DarkFlamex1** [comment]
Okay I was working on the networking and I hit a few problems.
I guess we are not doing singleplayer are we?
Anyways to make a better easier smoother multiplayer, the player controller should use character controller not rigidbodies.
And, the main contoller should just genereate target coordinates for rotation and position which will sync over network.

**Comment by 64bitpandas** [comment]
OK, that sounds good. CharacterController seems much simpler than
rigidbodies.
Also, character controller does not have gravity, right?

And what would be the best way to log the coordinates?

**Comment by EnumC** [comment]
Character Controller does not have gravity. You have to manually move it to simulate gravity. Here is a question about it: http://answers.unity3d.com/questions/334708/gravity-with-character-controller.html

Also, @DarkFlamex1 , why do the controller have to generate the coordinates? Can't we just simply drag transform to observe?

**Comment by DarkFlamex1** [comment]
I was thinking we can predict the position or something else and send that
to a server object which calculates and does the movement.

**Comment by DarkFlamex1** [comment]
Answer to second question, yah I guess we can I am sleepy and can't think
straight I will see tmrw.

**Comment by DarkFlamex1** [comment]
Okay, I want to get something straight, are we going to code in a way to protect from hacking? Or will we just do it assuming no hackers? Cause if we just observer transform from client, a hacker can easily move around the map.

**Comment by DarkFlamex1** [comment]
So yes o no?d

**Comment by 64bitpandas** [comment]
Assume no hackers

**Comment by EnumC** [comment]
@DarkFlamex1 Agreed. Just assume that there is no hackers. We can improve on it later.

**Comment by DarkFlamex1** [comment]
Okay, Um is there any way that we can get all the player scripts on the parent or something?

**Comment by DarkFlamex1** [comment]
Guess what, PLAYERS CAN SYNC POSITIONS!

**Comment by 64bitpandas** [comment]
Nice!
I'll just fix a few small things and commit soon.
Also everyone please update Unity. Thanks :)

**Comment by EnumC** [comment]
@BenCuan @DarkFlamex1 @akhilanganesh @CrispyCookieHD @DohyunK 
Open Unity, load the project. On the top bar, click "Help" -> "About Unity" On the top, if it doesn't say "Version 5.3.4f1" or higher, that means your build is outdated. Please download the new installer, and install the newest version. If you haven't been added to the organization, or the group chat yet, leave a message, and we will add you. Thanks. :)
Also, Please do `git fetch origin develop` , then `git reset --hard origin/develop` . If your local repo is outdated. And those 2 commands will ignore all local changes, and get the newest commit. If for some reason those 2 doesn't work, do `git pull --rebase origin develop`

**Comment by DarkFlamex1** [comment]
Group chat?

**Comment by DarkFlamex1** [comment]
Oh btw that controller in the network player is temporary till we make a
new one !

**Comment by 64bitpandas** [comment]
I fixed the map and player to use the ones we've been working on. It works
pretty much the same as the testmap/testplayer Vikram made (I think). I
also re-enabled the GUI. Since the select teams buttons aren't done, just
press the escape key to start playing.

**Comment by 64bitpandas** [comment]
So guys, it seems like we are nearing a good point for the 1.1 release!
Let's finalize the list of features we want to include for this release and
start fixing some of the bugs as well. Please reply if you have any more
things that I forgot.
So far, the changelog:
-Added basic multiplayer setup
-Added teams (WIP still)
-Fixed jumping
-Added build checking/travis CI support
-Added fall detection

**Comment by DarkFlamex1** [comment]
Sounds Good!

**Comment by EnumC** [comment]
@BenCuan Just a thought, wouldn't it be better to get the game working? For example, how the towers will deduct x amount of xp. And how the players can win/score points, etc?

**Comment by 64bitpandas** [comment]
https://docs.google.com/document/d/1kZSpn4D0MxJK35Z2sqhvaWvhNsbz_B_9vobg80GeOm4/edit?usp=sharing

**Comment by 64bitpandas** [comment]
1.1 is COMPLETE!
Next step: Final bug fixes/testing to put in 1.1.1.
After that: New stuff to put in 1.2.
